feature/amazon/src/main/java/com/revenuecat/purchases/amazon/purchasing/ProxyAmazonBillingDelegate.kt

```diff
@@ -0,0 +1,64 @@
+package com.revenuecat.purchases.amazon.purchasing
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.os.ResultReceiver
+import androidx.annotation.VisibleForTesting
+import com.amazon.device.iap.model.RequestId
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.amazon.AmazonStrings
+import com.revenuecat.purchases.amazon.ProxyAmazonBillingActivity
+import com.revenuecat.purchases.amazon.ProxyAmazonBillingActivityBroadcastReceiver
+import com.revenuecat.purchases.amazon.PurchasingServiceProvider
+import com.revenuecat.purchases.common.errorLog
+
+internal class ProxyAmazonBillingDelegate {
+
+    @JvmSynthetic
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var broadcastReceiver: ProxyAmazonBillingActivityBroadcastReceiver? = null
+    private val filter = ProxyAmazonBillingActivityBroadcastReceiver.newPurchaseFinishedIntentFilter()
+
+    fun onCreate(activity: Activity, savedInstanceState: Bundle?) {
+        broadcastReceiver = ProxyAmazonBillingActivityBroadcastReceiver(activity)
+        activity.applicationContext.registerReceiver(broadcastReceiver, filter)
+        if (savedInstanceState == null) {
+            val requestId = startAmazonPurchase(activity.intent)
+            if (requestId == null) {
+                activity.finish()
+            }
+        }
+    }
+
+    fun onDestroy(activity: Activity) {
+        activity.applicationContext.unregisterReceiver(broadcastReceiver)
+        broadcastReceiver = null
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun startAmazonPurchase(intent: Intent): RequestId? {
+        val sku = intent.getStringExtra(ProxyAmazonBillingActivity.EXTRAS_SKU)
+        val resultReceiver =
+            intent.getParcelableExtra<ResultReceiver>(ProxyAmazonBillingActivity.EXTRAS_RESULT_RECEIVER)
+        val purchasingServiceProvider =
+            intent.getParcelableExtra<PurchasingServiceProvider>(
+                ProxyAmazonBillingActivity.EXTRAS_PURCHASING_SERVICE_PROVIDER
+            )
```

Comment:
I was wondering whether getting the arguments from the intent could be done in the activity... But I guess it's fine here so we have that logic available for testing here as well 👍.